nexus/test-utils/src/lib.rs

```diff
@@ -604,18 +636,21 @@ impl<'a, N: NexusServer> ControlPlaneTestContextBuilder<'a, N> {
         dns_config_client.dns_config_put(&dns_config).await.expect(
             "Failed to send initial DNS records to internal DNS server",
         );
-        dns_config
+        self.dns_config = Some(dns_config);
```

Comment:
This is just to avoid a `move` between setup steps, yeah?

Reply:
With this pattern, we no longer have a way to return a value that the next step can use. This is the easiest way to work around that (and follows the pattern the rest of the builder uses anyway).

I ran into something similar with the update-engine and came up with the design to use step handles that can be passed around, but only resolved once steps start executing. For example, https://github.com/oxidecomputer/omicron/blob/45b665184ad6254f5b2bb969abefbf476e0803c8/update-engine/examples/update-engine-basic/main.rs#L308-L310. But that's trying to do a lot more than we need here.